src/commons/utils/Hooks.ts

```diff
@@ -77,3 +77,26 @@ export function useInput<T>(defaultValue: T) {
     }
   };
 }
+
+/**
+ * This hook usage is similar to React.useState, the only difference
+ * being that the state is also written to local storage at the specified key on state updates.
+ *
+ * When calling this hook, the value will take on the stored value in local storage (if any).
+ * If this key-value does not exist in local storage yet, the default value will be used.
+ */
+export function useLocalStorageState<T>(
+  key: string,
+  defaultValue: T
+): [T, React.Dispatch<React.SetStateAction<T>>] {
+  const [value, setValue] = React.useState<T>(() => {
+    const localStorageValue = window.localStorage.getItem(key);
+    return localStorageValue ? JSON.parse(localStorageValue) : defaultValue;
```

Comment:
Consider catching any parsing errors (in case the local storage value is modified)

Reply:
I suppose a parse error would only arise if the local storage were changed manually, somehow. Maybe it's not so important to worry about such a case.